src/v/redpanda/admin_server.cc

```diff
@@ -3316,6 +3317,63 @@ void admin_server::register_self_test_routes() {
       });
 }
 
+ss::future<ss::json::json_return_type> admin_server::get_raft_metrics_handler(
+  std::unique_ptr<ss::httpd::request> req) {
+    const model::ntp ntp = parse_ntp_from_request(req->param);
+
+    if (need_redirect_to_leader(ntp, _metadata_cache)) {
+        throw co_await redirect_to_leader(*req, ntp);
+    }
+
+    seastar::httpd::debug_json::raft_metrics ans;
+
+    const bool is_controller = ntp == model::controller_ntp;
+    cluster::consensus_ptr raft;
+
+    if (!is_controller) {
+        auto partition = _partition_manager.local().get(ntp);
+        if (!partition) {
+            throw ss::httpd::not_found_exception(
+              fmt::format("Could not find ntp: {}", ntp));
+        }
+
+        raft = partition->raft();
+    } else {
+        raft = _controller->get_raft0();
+    }
+
+    if (!raft->is_leader()) {
+        throw ss::httpd::server_error_exception(
```

Comment:
this makes sense provided that `need_redirect_to_leader` also handles the leadership check for the controller. i took a quick peek, but wasn't sure it did. wdyt?

Reply:
`metadata_cache` contain info about controller topic. But as understand we can get situation where leader was not elected (only pre-vote). But `metadata_cache` contains already info about expected leader (So we will do dispatch), but consensus does not finish election. 

I am not sure for 100% how does it work, @mmaslankaprv could you help please

Reply:
How I understand it
`metadata_cache` knows that node is leader for partition, but consensus - not
Becasue `meatadata_cache` on starting redpanda contains node for leader that will be leader on 99.9%, but is not elected.